pkg/auth/requests/authenticate.go

```diff
@@ -54,11 +54,15 @@ func ToAuthMiddleware(a Authenticator) auth.Middleware {
 	return auth.ToMiddleware(auth.AuthenticatorFunc(f))
 }
 
-type ClusterRouter func(req *http.Request) string
+type (
+	ClusterRouter func(req *http.Request) string
+	userRefresher func(string, bool)
```

Comment:
This would be helped by naming the params (as is done above)?

You possibly don't even need to do this, you could just declare the only use inlined into the struct?

```go
type tokenAuthenticator struct {
    ...
    refreshUser         func(userID string, force bool)
}
```

Reply:
Agreed, never hurts. Done.